core/services/GeoSvc.cc

```
@@ -0,0 +1,677 @@
+
+  auto saveHeadPhsp = m_configSvc->GetValue<VecG4doubleSPtr>("GeoSvc", "SavePhSpHead");
+  auto saveUsrPhsp = m_configSvc->GetValue<VecG4doubleSPtr>("GeoSvc", "SavePhSpUsr");
+
+  if (G4String(savePhspRequest).compare("Usr:0.") != 0) { // differ from default value
+
+    // Clear vectors (of default positions) when SavePhspRequest string is not empty
+    saveHeadPhsp->clear();
+    saveUsrPhsp->clear();
+
+    // get rid of empty cells from the string value
+    std::remove_if(savePhspRequest.begin(), savePhspRequest.end(), [](unsigned char x) { return std::isspace(x); });
```

Cppcheck (2.19.0)

[warning] Return value of std

(uselessCallsRemove)